packages/strapi/lib/core/configurations.js

```diff
@@ -321,9 +322,13 @@ module.exports.app = async function() {
   // default settings
   this.config.port = get(this.config.currentEnvironment, 'server.port') || this.config.port;
   this.config.host = get(this.config.currentEnvironment, 'server.host') || this.config.host;
+  this.config.url = `http://${this.config.host}:${this.config.port}`;
```

Comment:
Baddddd, really shouldn't be hard coding this even in development :stuck_out_tongue:

Reply:
Check two lines before :) based on env